vassal-app/src/main/java/VASSAL/counters/Decorator.java

```diff
@@ -749,6 +777,12 @@ public static Command putOldProperties(GamePiece p) {
     comm = comm.append(container.setPersistentProperty(BasicPiece.OLD_BOARD, boardName));
     comm = comm.append(container.setPersistentProperty(BasicPiece.OLD_ZONE, zoneName));
     comm = comm.append(container.setPersistentProperty(BasicPiece.OLD_LOCATION_NAME, locationName));
+    if (wasOnMat) {
+      comm = comm.append(container.setPersistentProperty(BasicPiece.OLD_MAT, matName));
+      comm = comm.append(container.setPersistentProperty(BasicPiece.OLD_MAT_ID, matID));
+      comm = comm.append(container.setPersistentProperty(BasicPiece.OLD_MAT_PIECE_NAME, matPieceName));
+      comm = comm.append(container.setPersistentProperty(BasicPiece.OLD_MAT_BASIC_NAME, matBasicName));
+    }
```

Comment:
Do the OLD Mat properties ever get reset to null once a piece is no longer on a mat? Should the next move where !wasOnMat set them to null?

Reply:
Ah yes you're right, I out-clevered myself. Fixed now.